tests/check/check_test.go

```diff
@@ -250,7 +250,8 @@ type document
 			require.NotEmpty(t, fields["trace_id"])
 			if !test.expectedError {
 				require.NotEmpty(t, fields["datastore_query_count"])
-				require.Len(t, fields, 13)
+				require.NotNil(t, fields["dispatch_count"])
```

Comment:
The dispatch count for this specific example is zero so asserting `require.NotEmpty` would result in an error.

Reply:
nit (non-blocking) - If you're consuming this field you probably don't want it to change from numeric to string accidentally as well, so it may be worth adding that type assertion on the value type as well 🤷 ? This in addition to the `NotNil` check already here.